src/Framework/MockObject/Generator/Generator.php

```diff
@@ -415,21 +410,17 @@ public function generate(string $type, bool $mockObject, bool $markAsMockObject,
             serialize($callOriginalMethods),
         );
 
-        if (!isset(self::$cache[$key])) {
-            self::$cache[$key] = $this->generateCodeForTestDoubleClass(
-                $type,
-                $mockObject,
-                $markAsMockObject,
-                $methods,
-                $mockClassName,
-                $callOriginalClone,
-                $callAutoload,
-                $cloneArguments,
-                $callOriginalMethods,
-            );
-        }
-
-        return self::$cache[$key];
+        return $this->generateCodeForTestDoubleClass(
```

Comment:
> I noticed that (counter-intuitively) memory consumption increases when test double code declarations are not cached.

sounds like this method is somehow triggering logic which when invoked repeatedly builds up a buffer or uses a reasource which leaks memory.

Reply:
when running blackfire on 778070d58, I don't see `generateCodeForTestDoubleClass` at all in the profile -> its using very little memory and therefore blackfire decides its not worth having information about this method in the profile.

when running blackfire on this PR, we get a bit more information. `generateCodeForTestDoubleClass` is still using very little memory, but at least it now shows up in the profile.

in this screen you can see how the memory used by `generateCodeForTestDoubleClass` spreads over calls triggered from this method:

![grafik](https://github.com/sebastianbergmann/phpunit/assets/120441/c7193a05-fa71-459b-a451-b541068787e2)

it looks like the lions share of memory is consumed by autoloading and reflection, which makes sense to me.
both are apis which have side-effects.

if there is a way to reduce/remove calls to these 2 php-src apis you might be able to remove the caching.

----

I ran blackfire on the unit test-suite as a whole. in case we could find a command in which the test-double generator itself would take a bigger share of the overall memory we could get more precise numbers of the diff between main and this PR.